econml/sklearn_extensions/model_selection.py

```diff
@@ -190,3 +201,186 @@ def split(self, X, y, sample_weight=None):
             Weights associated with the training data.
         """
         return _split_weighted_sample(self, X, y, sample_weight, is_stratified=True)
+
+
+def cross_val_predict(estimator, X, y=None, *, groups=None, cv=None,
```

Comment:
I would make this a private function and remove most of the docstring and instead add a reference to the sklearn method. This signals to the user that this method might change in the future (if, say, the sklearn one changes) and they shouldn't use it as is.

Reply:
Good point!